src/dialects/mysql/schema/charset-handler.js

```diff
@@ -0,0 +1,167 @@
+import _ from 'lodash';
+
+export function getCharsetAndCollation(single, connectionSettings, driverName) {
+  if (single.charset) {
+    return {
+      charset: single.charset,
+      collation: single.collate || '',
+    };
+  }
+  if (!connectionSettings.charset) {
+    return {
+      charset: '',
+      collation: '',
+    };
+  }
+
+  return getMappedCharsetAndCollation(
+    connectionSettings.charset.toUpperCase(),
+    driverName
+  );
+}
+
+function getMappedCharsetAndCollation(charset, driverName) {
+  // This is the map of what the mysql libraries calls charset, which is either
+  // character sets which they map to default collations or collations themselves
+  let charsetMap;
+  let aliases;
+  try {
+    if (driverName === 'mysql') {
+      charsetMap = require('mysql/lib/protocol/constants/charsets');
+      aliases = mysqlAliases;
+    } else if (driverName === 'mysql2') {
```

Comment:
Probably this thing should have been broken in two parts, where aliasesList is overridden in each dialect version and not needing to have this if driverName is something hack in base implementation.